src/autobridgecalculator.js

```diff
@@ -46,7 +46,27 @@ AutobridgeCalculator.prototype.calculate = function() {
 
   this.clearOwnerFundsLeftover();
 
+  this._calculateInternal(legOnePointer, legTwoPointer, offersAutobridged,
+    callback);
+};
+
+AutobridgeCalculator.prototype._calculateInternal = function(
+  legOnePointer, legTwoPointer, offersAutobridged, callback = function() {}
+) {
+
+  const startTime = Date.now();
+
   while (this.legOneOffers[legOnePointer] && this.legTwoOffers[legTwoPointer]) {
+    const lasted = (Date.now() - startTime);
```

Comment:
Why do we need to care about the time? Also, I don't understand the purpose of this block of code, can you explain more?